compute/src/main/java/org/jclouds/compute/domain/internal/ArbitraryCpuRamTemplateBuilderImpl.java

```diff
@@ -46,9 +46,12 @@ protected ArbitraryCpuRamTemplateBuilderImpl(@Memoized Supplier<Set<? extends Lo
       super(locations, images, hardwares, defaultLocation, optionsProvider, defaultTemplateProvider);
    }
 
-   protected Hardware automaticHardwareForCpuAndRam(double cores, int ram) {
-      return new HardwareBuilder()
-            .id(automaticHardwareIdSpecBuilder(cores, ram).toString())
+   protected Hardware automaticHardwareForCpuAndRam(double cores, int ram, float diskSize) {
```

Comment:
Also, the "disk" parameter is optional. Better use a `Float` object to allow to pass null, or even better an `Optional<Float>`, so callers know exactly what the contract of the method is.